### ISSUE TITLE
Unable to stage a specific hunk on macOS Monterey

### ISSUE THREAD

**Issue by ghost** [extrawurst/gitui#1561]
#### Describe the bug
Unable to stage a specific hunk, but can stage an entire file.

#### To Reproduce
1. In the "Diff" view, press "Enter" on a hunk.
3. See that the progress spinner appears for a half second.
4. See that nothing is staged, the diff view is unaltered.
5. In the "Unstaged Changes" view, press "Enter" on the file that contained the hunk.
6. See that the entire file is staged without issue. 

#### Expected behavior
The hunk should stage when enter is pressed while it is select in the "Diff" view.

#### Context
 - OS/Distro + Version: `macOS 12.6.1`
 - GitUI Version: `0.22.1`

**Comment by adam-becker** [comment]
Some additional info: If I try and reset a hunk, gitui crashes with a `Hunk not found` error.

**Comment by conor-byrne** [comment]
I am also having this issue on Windows 10, gitui version 0.22.1

gitui.log:
```
13:28:14 [TRACE] (1) gitui::app: [src\app.rs:382] event: Input(Key(KeyEvent { code: Down, modifiers: NONE, kind: Press, state: NONE }))
13:28:14 [TRACE] (1) gitui::app: [src\app.rs:382] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE, kind: Press, state: NONE }))
13:28:14 [TRACE] (1) gitui::app: [src\app.rs:477] update
13:28:14 [TRACE] (1) asyncgit::diff: [asyncgit\src\diff.rs:98] request DiffParams { path: "controller/get1032/Configure001.c", diff_type: WorkDir, options: DiffOptions { ignore_whitespace: true, context: 4, interhunk_lines: 1 } }
13:28:14 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:102] request: [hash: 7363449280220696857] (type: WorkingDir)
13:28:14 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:102] request: [hash: 17980392194271790495] (type: Stage)
13:28:14 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Diff)
13:28:14 [TRACE] (5) asyncgit::status: [asyncgit\src\status.rs:160] status fetched: 17980392194271790495 (type: Stage)
13:28:14 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Status)
13:28:14 [TRACE] (3) asyncgit::status: [asyncgit\src\status.rs:160] status fetched: 7363449280220696857 (type: WorkingDir)
13:28:14 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Status)
13:28:16 [DEBUG] (8) gitui::watcher: notify events: 5
13:28:16 [DEBUG] (8) gitui::watcher: notify [0]: DebouncedEvent { path: "E:/work/.git\\index", kind: AnyContinuous }
13:28:16 [DEBUG] (8) gitui::watcher: notify [1]: DebouncedEvent { path: "E:/work/.git\\objects\\tmp_object_git2_82cc90ef7118d582", kind: Any }
13:28:16 [DEBUG] (8) gitui::watcher: notify [2]: DebouncedEvent { path: "E:/work/.git\\index.lock", kind: AnyContinuous }
13:28:16 [DEBUG] (8) gitui::watcher: notify [3]: DebouncedEvent { path: "E:/work/.git\\objects\\7c\\991631b0443cfe007c5e376edec4963dbe9320", kind: AnyContinuous }
13:28:16 [DEBUG] (8) gitui::watcher: notify [4]: DebouncedEvent { path: "E:/work/.git", kind: AnyContinuous }
13:28:16 [TRACE] (1) gitui::app: [src\app.rs:477] update
13:28:16 [TRACE] (1) asyncgit::diff: [asyncgit\src\diff.rs:98] request DiffParams { path: "controller/get1032/Configure001.c", diff_type: WorkDir, options: DiffOptions { ignore_whitespace: true, context: 4, interhunk_lines: 1 } }
13:28:16 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:102] request: [hash: 10086405885517345526] (type: WorkingDir)
13:28:16 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:102] request: [hash: 9590567146677191896] (type: Stage)
13:28:16 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Diff)
13:28:16 [TRACE] (5) asyncgit::status: [asyncgit\src\status.rs:160] status fetched: 9590567146677191896 (type: Stage)
13:28:16 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Status)
13:28:16 [TRACE] (3) asyncgit::status: [asyncgit\src\status.rs:160] status fetched: 10086405885517345526 (type: WorkingDir)
13:28:16 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Status)
13:28:17 [DEBUG] (8) gitui::watcher: notify events: 5
13:28:17 [DEBUG] (8) gitui::watcher: notify [0]: DebouncedEvent { path: "E:/work/.git\\index.lock", kind: Any }
13:28:17 [DEBUG] (8) gitui::watcher: notify [1]: DebouncedEvent { path: "E:/work/.git\\objects\\tmp_object_git2_69498939c575b563", kind: Any }
13:28:17 [DEBUG] (8) gitui::watcher: notify [2]: DebouncedEvent { path: "E:/work/.git\\objects\\7c\\991631b0443cfe007c5e376edec4963dbe9320", kind: Any }
13:28:17 [DEBUG] (8) gitui::watcher: notify [3]: DebouncedEvent { path: "E:/work/.git", kind: Any }
13:28:17 [DEBUG] (8) gitui::watcher: notify [4]: DebouncedEvent { path: "E:/work/.git\\index", kind: Any }
13:28:17 [TRACE] (1) gitui::app: [src\app.rs:477] update
13:28:17 [TRACE] (1) asyncgit::diff: [asyncgit\src\diff.rs:98] request DiffParams { path: "controller/get1032/Configure001.c", diff_type: WorkDir, options: DiffOptions { ignore_whitespace: true, context: 4, interhunk_lines: 1 } }
13:28:17 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:102] request: [hash: 4555857745244010157] (type: WorkingDir)
13:28:17 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:102] request: [hash: 13658895302602991637] (type: Stage)
13:28:17 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Diff)
13:28:17 [TRACE] (4) asyncgit::status: [asyncgit\src\status.rs:160] status fetched: 13658895302602991637 (type: Stage)
13:28:17 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Status)
13:28:17 [TRACE] (3) asyncgit::status: [asyncgit\src\status.rs:160] status fetched: 4555857745244010157 (type: WorkingDir)
13:28:17 [TRACE] (1) gitui::app: [src\app.rs:496] update_async: Git(Status)
13:28:17 [TRACE] (1) gitui::app: [src\app.rs:382] event: Input(Key(KeyEvent { code: Char('c'), modifiers: CONTROL, kind: Press, state: NONE }))
13:28:17 [ERROR] input thread error: sending on a disconnected channel
```

**Comment by extrawurst** [comment]
will be fixed in #1746
